src/lib/swapper/swappers/ThorchainSwapper/utils/addAggregatorAndDestinationToMemo.ts

```diff
@@ -41,39 +42,21 @@ export const addAggregatorAndDestinationToMemo = ({
     affiliateBps,
   ] = quotedMemo.split(MEMO_PART_DELIMITER)
 
+  const maxMemoSize = (() => {
+    if (sellChainId === bchChainId) return BCH_MAXIMUM_BYTES_LENGTH
+    if (isUtxoChainId(sellChainId)) return UTXO_MAXIMUM_BYTES_LENGTH
+    return Infinity
+  })()
+
   const finalAssetLimitWithManualSlippage = subtractBasisPointAmount(
     bn(minAmountOut).toFixed(0, BigNumber.ROUND_DOWN),
     slippageBps,
     BigNumber.ROUND_DOWN,
   )
 
-  const maximumPrecision = 6
-  const endingExponential = finalAssetPrecision - maximumPrecision
-  const finalAssetLimitCryptoPrecision = fromBaseUnit(
-    finalAssetLimitWithManualSlippage,
-    finalAssetPrecision,
-    maximumPrecision,
-  )
-  const shouldPrependZero = endingExponential < 10
-  const thorAggregatorExponential = shouldPrependZero
-    ? `0${endingExponential > 0 ? endingExponential : '1'}`
-    : endingExponential
-
-  // The THORChain aggregators expects this amount to be an exponent, we need to add two numbers at the end which are used at exponents in the contract
-  // We trim 10 of precisions to make sure the THORChain parser can handle the amount without precisions and rounding issues
-  // If the finalAssetPrecision is under 5, the THORChain parser won't fail and we add one exponent at the end so the aggregator contract won't multiply the amount
-  const finalAssetLimitWithTwoLastNumbersAsExponent = `${
-    finalAssetPrecision < maximumPrecision
-      ? finalAssetLimitWithManualSlippage
-      : finalAssetLimitCryptoPrecision.replace('.', '')
-  }${thorAggregatorExponential}`
-
-  // Paranoia assertion - expectedAmountOut should never be 0 as it would likely lead to a loss of funds.
-  assert(
-    BigInt(finalAssetLimitWithTwoLastNumbersAsExponent) > 0n,
-    'expected finalAssetLimitWithManualSlippage to be a positive amount',
-  )
+  const finalAssetAddressShortened = getUniqueAddressSubstring(finalAssetAddress, longtailTokens)
 
+  // THORChain themselves uses 2 characters but it might collide at some points in the future (https://gitlab.com/thorchain/thornode/-/blob/develop/x/thorchain/aggregators/dex_mainnet_current.go)
```

Comment:
Just commenting here so @0xApotheosis doesn't have to
```suggestion
  // THORChain themselves use 2 characters but it might collide at some point in the future (https://gitlab.com/thorchain/thornode/-/blob/develop/x/thorchain/aggregators/dex_mainnet_current.go)
```

Reply:
Me baguette